pootle/apps/pootle_store/updater.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from django.utils import timezone
+from django.utils.functional import cached_property
+
+from pootle_store.constants import POOTLE_WINS
+
+
+class StoreUpdate(object):
+
+    def __init__(self, source, **kwargs):
+        self.source = source
+        self.kwargs = kwargs
+
+    @property
+    def uids(self):
+        return list(self.kwargs["uids"])
+
+    @property
+    def change_indeces(self):
+        return self.kwargs["change_indeces"]
+
+    @property
+    def indeces(self):
+        return self.kwargs["indeces"]
+
+    @property
+    def store_revision(self):
+        return self.kwargs["store_revision"]
+
+    @property
+    def submission_type(self):
+        return self.kwargs["submission_type"]
+
+    @property
+    def update_revision(self):
+        return self.kwargs["update_revision"]
+
+    @property
+    def resolve_conflict(self):
+        return self.kwargs["resolve_conflict"]
+
+    @property
+    def user(self):
+        return self.kwargs["user"]
+
+
+class FrozenUnit(object):
+
+    def __init__(self, unit):
+        self.target = unit.target_f
+        self.state = unit.state
+        self.submitter = unit.submitted_by
+        self.translator_comment = unit.translator_comment
+
+
+class UnitUpdate(object):
+
+    def __init__(self, unit, target_store, update):
+        self.unit = unit
+        self.target_store = target_store
+        self.update = update
+        self.original = FrozenUnit(unit)
+
+    @property
+    def translator_comment(self):
+        return self.unit.getnotes(origin="translator")
+
+    @property
+    def comment_updated(self):
+        return (
+            self.original.translator_comment != self.translator_comment
+            and (self.original.translator_comment or self.translator_comment))
```

Comment:
Doesn't it make sense to alter the order and check first if these comments are empty or exist before comparing them?